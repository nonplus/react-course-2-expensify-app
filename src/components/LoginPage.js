import { Link } from "react-router-dom";
import { connect } from "react-redux";
import React from "react";
import { loginAsync } from "../actions/auth";

export const LoginPage = ({ loginAsync }) => (
  <div>
    <h1>Login</h1>
    <p>
      <button onClick={loginAsync}>Login</button>
    </p>
  </div>
);

const mapDispatchToProps = dispatch => ({
  loginAsync: () => dispatch(loginAsync())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
