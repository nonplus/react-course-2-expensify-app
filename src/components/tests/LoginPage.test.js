import { shallow } from "enzyme";
import React from "react";
import { LoginPage } from "../LoginPage";

describe("<LoginPage />", () => {
  let wrapper, loginAsync;
  beforeEach(() => {
    loginAsync = jest.fn();
    wrapper = shallow(<LoginPage loginAsync={loginAsync} />);
  });

  it("should render page correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("<button />", () => {
    let button;
    beforeEach(() => (button = wrapper.find("button")));

    describe("onClick", () => {
      it("should call props.loginAsync", () => {
        button.simulate("click");
        expect(loginAsync).toHaveBeenCalled();
      });
    });
  });
});
