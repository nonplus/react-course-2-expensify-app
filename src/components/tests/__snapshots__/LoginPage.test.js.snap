// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> should render page correctly 1`] = `
<div>
  <h1>
    Login
  </h1>
  <p>
    <button
      onClick={[Function]}
    >
      Login
    </button>
  </p>
</div>
`;
